src/nodes.ts

```diff
@@ -1525,6 +1541,25 @@ export class NewOp extends Node {
   }
 }
 
+export class SoakedNewOp extends Node {
+  readonly ctor: Node;
+  readonly arguments: Array<Node>;
+
+  constructor(
+    line: number,
+    column: number,
+    start: number,
+    end: number,
+    raw: string,
+    ctor: Node,
+    args: Array<Node>
```

Comment:
Ah, good to know, I didn't notice the difference. Although looks like that doesn't work in this case since the field name is `arguments` and you can't have that as a parameter name. Might be nice to use `args` as the property name everywhere for function invocation nodes, but I'll call that out of scope for this change.